src/lib/utils/createPreinstalledProxy.js

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2025 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+// eslint-disable-next-line import/extensions
+import { poll } from "./poll.js";
+
+/**
+ * Creates a proxy function for a preinstalled Alloy instance.
+ * The proxy waits for the external instance to be loaded and configured
+ * before forwarding commands to it.
+ *
+ * @param {String} name - The instance name to look for on window
+ * @param {Object} [options={}] - Configuration options
+ * @param {Number} [options.timeout=1000] - Max time to wait for instance (ms)
+ * @param {Number} [options.interval=100] - Polling interval (ms)
+ * @param {Number} [options.configTimeout=5000] - Max time to wait for getLibraryInfo (ms)
+ * @param {Function} [options.onWarn] - Warning callback (msg) => void
+ * @param {Function} [options.onError] - Error callback (msg) => void
+ * @return {Function} - Proxy function that forwards to real instance
+ */
+export const createPreinstalledProxy = (name, options = {}) => {
```

Comment:
This creates a "proxy" instance that will poll for the real preinstalled instance on the `window` object.
The goal is to address possible race conditions between the loading of the pre installed instance against the launch container.